### ISSUE TITLE
SOCAT problem

### ISSUE THREAD

**Issue by Fresco-Bis** [DCSFlightpanels/dcs-bios#160]
I'm very new to this so I'm not sure this is the problem but after hours of reinstalling this and that and trying solutions for that and this I can only come to the conclusion this is the problem. I have followed every step from multiple sources (YouTube and official installation guides provided by the documentation) for installing and it just does not work. When I open the connect-serial-port.cmd and enter my com port- 3- all it does is bring up the device's status and then gives an error message, afterwards closing the program.

connect-serial-port.cmd:

Enter a COM Port Number:3

Status for device COM3:
-----------------------
    Baud:            250000
    Parity:          None
    Data Bits:       8
    Stop Bits:       1
    Timeout:         OFF
    XON/XOFF:        OFF
    CTS handshaking: OFF
    DSR handshaking: OFF
    DSR sensitivity: OFF
    DTR circuit:     OFF
    RTS circuit:     OFF

Waiting for 4 seconds, press a key to continue ...
2023/04/08 15:25:31 socat[784] E setsockopt(5, 0, 12, {0x0a32ffef,0x00000000}, 8): No route to host
Press any key to continue . . .

(After hitting enter at this point again it will then automatically close the cmd)

I have tried using both socat 32bit and 64bit version. I have tried using ensure-socat which does nothing by what I can see. I also tried to meddle with DcsBiosCOMHandler, needless to say I couldn't understand that or get it to work so I gave up there. I cant even run connect-serial-port in administrator because then it will say it cant "ensure-socat.cmd" location. I've made it so the connect-serial-port automatically selected 3. I have tried turning off all my firewall settings and nothings changed except now I have to deal with all its new antics. 

if it helps at all I use the steam version of DCS and I have it set to open beta as well (I only bring this up because I know the file names and maybe some other stuff is different between the two). If anyone can help I'd very much so appreciate it.

**Comment by WarLord211** [comment]
instalation and folder management is OK?

https://github.com/DCSFlightpanels/dcs-bios/blob/master/Scripts/DCS-BIOS/doc/DCS-BIOS-FLIGHTPANELS_Install%20Guide.pdf

**Comment by Fresco-Bis** [comment]
I believe so. To be on the safe side, I deleted all the files pertaining to DCS-BIOS and DCSFP (originally I got the old DCS-BIOS then found out about DCSFP so still had some of those old files, DCS-HUB never worked either) so I could start over, then reinstalled DCSFP. I followed the install guide you linked (Which I had used before), but I made sure to double check everything was done correctly and I still get the same issue. I then went into the trouble shooting guide and didn't find much success there. 

Specifically, the part about changing the IP and Port in the UDP communications, I thought it could be a fix for my problem from what I was understanding but didn't change anything. Granted, I didn't try all combinations of IP's and Port's on offer but if a few different sets didn't work I decided to stop trying that route (I made sure to update each files IP and Port after changing them). I did have Hamachi, however I'm sure I had completely disabled it, just to be sure I completely uninstalled it, still no change. 

The last thing I can think of is possibly a firewall problem? But previously I had completely turned it off during my first attempts, maybe I didn't fully turn it off or allowed firewall to access, I'll look more into it after sending this. 

This may not help or it may I'm not really sure, but when checking with the DCS-BIOS Control Reference, it cant find and connect to DCS World either. I thought it was strange as I had originally used BORT before hand and had the same issue but it didn't out right tell me. It says: DCS Connection: Connection failed. Refresh the page to try again. Once again, while DCS is running, it will never connect to it. No matter how many times a refresh it and set it to live data.

**Comment by WarLord211** [comment]
socat cant work if BIOS doestn work correct.
first BIOS must work via Control Ref or BORT.
if you see here valves then socat do something.

if you installed evrything right (like in the Guide) maybe there are conflicts in the export.lua

**Comment by Fresco-Bis** [comment]
The export.lua isn't the problem, its correct as is and I made sure to double check with text from the troubleshooting guide. 

Some other users on the discord told me to try and alter my windows defender and firewall setting, I've added the socat and connect-serial-port.cmd as exceptions from the windows antivirus/protection software but it hasn't changed anything, I've even committed to fully turning off windows defender and firewall. 

Yet it still gives the error message: 
2023/04/11 23:28:05 socat[7] E setsockopt(5, 0, 12, {0x0a32ffef,0x00000000}, 8): No route to host
Press any key to continue . . .

by looking at the UDP connections trouble shooting guide, there is a range of Ports and IP's to use. are there any "specific" IP's and Ports to try and use rather than being given a range? I'm not sure if that would help or fix the problem, but it's just something I though of.

**Comment by Fresco-Bis** [comment]
That is just unbelievable, a while ago I uninstalled Hamachi, but turns out like you said that network adaptor was still there preventing everything from working... somehow lol. after deleting that it worked perfectly fine. Hamachi was great for use with ARMA 2 with friends, but definitely not worth all I just went through XD. I'm so sorry for making you go through so many days of helping me lol. From the bottom of my heart WarLord, thank you for enduring me and helping me hahaha, I'll make sure to use DCSFP and make awesome things with it. Keep it up with the awesome work from now and hopefully far into the future :)
